src/api-binder/report.ts

```diff
@@ -70,30 +76,44 @@ async function report({ body, opts }: StateReport) {
 	}
 }
 
+/**
+ * Collects current state and reports with backoff. Diffs report content with
+ * previous report and does not send an empty report.
+ *
+ * Does *not* validate time elapsed since last report.
+ */
 async function reportCurrentState(opts: StateReportOpts) {
-	// Wrap the report with fetching of state so report always has the latest state diff
 	const getStateAndReport = async () => {
-		const now = performance.now();
-		// Only try to report if enough time has elapsed since last report
-		if (now - lastReportTime >= constants.maxReportFrequency) {
-			const currentState = await deviceState.getCurrentForReport(lastReport);
-			const stateDiff = prune(shallowDiff(lastReport, currentState, 2));
+		const currentState = await deviceState.getCurrentForReport(lastReport);
+		const stateDiff = prune(shallowDiff(lastReport, currentState, 2));
 
-			if (empty(stateDiff)) {
+		if (empty(stateDiff)) {
+			return;
+		}
+
+		// If metrics not yet scheduled, report must include a state change to
+		// qualify for sending.
+		const metricsScheduled =
+			performance.now() - lastReportTime > maxMetricsFrequency;
+		if (!metricsScheduled) {
+			const uuid = await config.get('uuid');
+			if (!uuid) {
+				throw new InternalInconsistencyError('No uuid found for local device');
+			}
+			const uuidMap = stateDiff[uuid] as { string: any };
```

Comment:
Is this type correct? Shouldn't it be 

```suggestion
			const uuidMap = stateDiff[uuid] as { [k: string]: any };
```

Disregard if I'm misunderstanding

Reply:
TIL, thanks.